fedimint-core/src/config.rs

```diff
@@ -272,6 +284,40 @@ pub type CommonModuleGenRegistry = ModuleGenRegistry<DynCommonModuleGen>;
 /// distinct config for dkg.
 pub type ServerModuleGenParamsRegistry = ModuleGenRegistry<ConfigGenParams>;
 
+impl Eq for ServerModuleGenParamsRegistry {}
+
+impl PartialEq for ServerModuleGenParamsRegistry {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.eq(&other.0)
+    }
+}
+
+impl Serialize for ServerModuleGenParamsRegistry {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut serializer = serializer.serialize_map(Some(self.0.len()))?;
+        for (key, value) in self.0.iter() {
+            serializer.serialize_key(key)?;
+            serializer.serialize_value(&value.0)?;
+        }
+        serializer.end()
+    }
+}
+
+impl<'de> Deserialize<'de> for ServerModuleGenParamsRegistry {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let json: BTreeMap<ModuleKind, serde_json::Value> = Deserialize::deserialize(deserializer)?;
+        let mut params = BTreeMap::new();
+
+        for (key, value) in json {
+            params.insert(key, ConfigGenParams(value));
+        }
+        Ok(ModuleGenRegistry(params))
+    }
+}
```

Comment:
why not separately impl serde for `ModuleGenRegistry<T>` and `ConfigGenParams`

Reply:
I guess I could implement it for `ModuleGenRegistry<T: Serialize>`, but since this is the only place that uses it right now I didn't think it was necessary.